common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -418,6 +415,47 @@ static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2
         }
     }
 
+    static int equalsConstantTime(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2, int length) {
+        long result = 0;
+        final long baseOffset1 = BYTE_ARRAY_BASE_OFFSET + startPos1;
+        final long baseOffset2 = BYTE_ARRAY_BASE_OFFSET + startPos2;
+        final int remainingBytes = length & 7;
+        final long end = baseOffset1 + remainingBytes;
+        for (long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length; i >= end; i -= 8, j -= 8) {
+            result |= UNSAFE.getLong(bytes1, i) ^ UNSAFE.getLong(bytes2, j);
```

Comment:
Not sure how you would check, but being able to do unaligned access has a noticeble performance degradation.  Intel lets you do it, but just does it slower.  (i.e. has to load two cache lines instead of one, and takes longer to go through the pipeline).  

The benchmark numbers on the Go version saw a boost (~460ns -> 320ns comparing two 4k buffers)  by having correct alignment.  I have no idea how Java tells you if something is aligned, since I don't know how the memory allocator works.  Since java has a compacting GC, any pointers to data are an extra level of indirection away right?